packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -416,36 +416,129 @@ class DaemonDomain extends Domain {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have an iOS platform directory');
+            }
+          case PlatformType.android:
+            if (!featureFlags.isAndroidEnabled) {
+              isSupported = false;
+              reasons.add('the Android feature is not enabled ("flutter config --enable-android")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.android)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have an Android platform directory');
+            }
+          case PlatformType.web:
+            if (!featureFlags.isWebEnabled) {
+              isSupported = false;
+              reasons.add('the Web feature is not enabled ("flutter config --enable-web")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.web)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Web platform directory');
+            }
+          case PlatformType.fuchsia:
+            if (!featureFlags.isFuchsiaEnabled) {
+              isSupported = false;
+              reasons.add('the Fuchsia feature is not enabled ("flutter config --enable-fuchsia")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.fuchsia)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Fuchsia platform directory');
+            }
+          case PlatformType.custom:
+            if (!featureFlags.areCustomDevicesEnabled) {
+              isSupported = false;
+              reasons.add('the custom-devices feature is not enabled ("flutter config --enable-custom-devices")');
+              canBeEnabledWithCreate = false;
+            }
+          case PlatformType.windowsPreview:
+            // TODO(fujino): detect if there any plugins with native code
+            if (!featureFlags.isPreviewDeviceEnabled) {
+              isSupported = false;
+              reasons.add('the flutter-preview feature is not enabled ("flutter config --enable-flutter-preview")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.windows)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Windows platform directory');
+            }
+        }
+
+        if (isSupported) {
+          platforms.add(platform.name);
+          platformTypesMap[platform.name] = const <String, Object>{
+            'isSupported': true,
+          };
+        } else {
+          platformTypesMap[platform.name] = <String, Object>{
+            'isSupported': false,
+            'reason': reasons.join(' and '),
```

Comment:
Should we just send the array and let the client join them in this way if most appropriate? (for example depending on how it's rendered, the client might prefer a list than a long sentence).

Reply:
sounds good to me

Reply:
done